cylc/rose/entry_points.py

```diff
@@ -120,6 +122,10 @@ def get_rose_vars(srcdir=None, opts=None):
             raise NotARoseSuiteException()
         return config
 
+    # Check for definitely invalid defines
+    if opts and hasattr(opts, 'defines') and not cylc7_back_compat:
```

Comment:
Why are we skipping this check in `cylc7_back_compat` mode?

Reply:
This is a check that didn't exist at Cylc 7. We shouldn't modify the behaviour in compatibility mode.

Reply:
I think this would have resulted in an error at Cylc 7 just as it has at Cylc 8.

The exact error would be different, but either way, it wouldn't work with either. The point of this PR was to catch this easily detectable error early so I don't think that back-compat mode should come into it?

Reply:
@wxtim, I don't think the back-compat mode check makes any sense here:

* This is a CLI interface.
* Missing the `=` symbol will result in the option being ignored.